src/server/cluster/cluster_shard_migration.cc

```diff
@@ -0,0 +1,57 @@
+// Copyright 2023, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+#include "server/cluster/cluster_shard_migration.h"
+
+#include <absl/flags/flag.h>
+#include <absl/strings/str_cat.h>
+
+#include "base/logging.h"
+#include "server/error.h"
+
+ABSL_DECLARE_FLAG(int, source_connect_timeout_ms);
+
+namespace dfly {
+
+using namespace std;
+using namespace facade;
+using absl::GetFlag;
+
+ClusterShardMigration::ClusterShardMigration(ServerContext server_context, uint32_t shard_id)
+    : ProtocolClient(server_context), source_shard_id_(shard_id) {
+}
+
+ClusterShardMigration::~ClusterShardMigration() {
+}
+
+std::error_code ClusterShardMigration::StartSyncFlow(BlockingCounter sb, Context* cntx) {
+  using nonstd::make_unexpected;
+
+  RETURN_ON_ERR(ConnectAndAuth(absl::GetFlag(FLAGS_source_connect_timeout_ms) * 1ms, &cntx_));
+
+  VLOG(1) << "Sending on flow " << source_shard_id_;
+
+  std::string cmd = absl::StrCat("DFLYMIGRATE FLOW ", source_shard_id_);
+
```

Comment:
I understand the design doc is already written. But maybe we can somehow avoid adding lots of new commands and use our DFLY command? 

For example, we have almost the same for replication (But replication needs to be backwards compatible 😞)

Because currently we have DFLY, DFLYMIGRATE, DFLYCLUSTER, REPLTAKEOVER...

Reply:
Not sure what are you saying. Are you suggesting that the commands for cluster should be subcommands of DFLY and not subcommands of DFLYCLUSTER or DFLYMIGRATE? 
BTW I dont think we need the DFLYMIGRATE FLOW command

Reply:
We sadly can't use DFLY FLOW because it expects us to be registered as a replica

Reply:
I can remove DFLYMIGRATE, I've created it like an internode communication command.